packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -212,6 +221,17 @@ function findRoutesAndCreateOptimalPath(
     const adjustedFills: Fill[] = [];
     const totalRoutedAmount = BigNumber.sum(...allSourcesRustRoute);
 
+    const sourcePathIdMap = ({} as any) as SourcePathIdMap;
+    const getSourcePathId = (key: ERC20BridgeSource) => {
```

Comment:
We often have multiple source paths for a single source. E.g., A->B @ uniswap, A->C->B @ uniswap, etc.